src/EnergyPlus/Data/EnergyPlusData.hh

```diff
@@ -66,119 +68,95 @@ struct BaseGlobalStruct
-{
-    struct ExteriorLightUsage
-    {
-        // Members
-        std::string Name;          // Descriptive name -- will show on reporting
-        int SchedPtr;              // Can be scheduled
-        Real64 DesignLevel;        // Consumption in Watts
-        Real64 Power;              // Power = DesignLevel * ScheduleValue
-        Real64 CurrentUse;         // Use for this time step
-        int ControlMode;           // Control mode Schedule Only or Astronomical Clock plus schedule
-        bool ManageDemand;         // Flag to indicate whether to use demand limiting
-        Real64 DemandLimit;        // Demand limit set by demand manager [W]
-        bool PowerActuatorOn;      // EMS flag
-        Real64 PowerActuatorValue; // EMS value
-        Real64 SumConsumption;     // sum of electric consumption [J] for reporting
-        Real64 SumTimeNotZeroCons; // sum of time of positive electric consumption [hr]
-
-                                   // Default Constructor
-        ExteriorLightUsage()
-            : SchedPtr(0), DesignLevel(0.0), Power(0.0), CurrentUse(0.0), ControlMode(1), ManageDemand(false), DemandLimit(0.0),
-            PowerActuatorOn(false), SumConsumption(0.0), SumTimeNotZeroCons(0.0)
-        {
+    struct DataGlobal : BaseGlobalStruct {
+        // Data
+        bool AnnualSimulation = false;
+
+        // MODULE VARIABLE DECLARATIONS:
+        std::string DayOfSimChr = "0";       // Counter for days (during the simulation) (character -- for reporting)
+
+        // MODULE PARAMETER DEFINITIONS
+        static constexpr int EndZoneSizingCalc = 4;
+
+        void clear_state() override {
+            AnnualSimulation = false;
+            DayOfSimChr = "0";
+        }
+    };
+
+    struct ExteriorEnergyUseData : BaseGlobalStruct {
+
+        int NumExteriorLights = 0; // Number of Exterior Light Inputs
+        int NumExteriorEqs = 0;    // Number of Exterior Equipment Inputs
+        Array1D<ExteriorEnergyUse::ExteriorLightUsage> ExteriorLights;        // Structure for Exterior Light reporting
+        Array1D<ExteriorEnergyUse::ExteriorEquipmentUsage> ExteriorEquipment; // Structure for Exterior Equipment Reporting
+        std::unordered_map<std::string, std::string> UniqueExteriorEquipNames;
+        bool GetExteriorEnergyInputFlag = true; // First time, input is "gotten"
+        ExteriorEnergyUseData() : NumExteriorLights(0), NumExteriorEqs(0), GetExteriorEnergyInputFlag(true) {}
+
+        void clear_state() {
+            NumExteriorLights = 0;
+            NumExteriorEqs = 0;
+            ExteriorLights.deallocate();
+            ExteriorEquipment.deallocate();
+            UniqueExteriorEquipNames.clear();
+            GetExteriorEnergyInputFlag = true;
+        }
+    };
+
+    struct FansData : BaseGlobalStruct {
+        // constants
+        static constexpr int ExhaustFanCoupledToAvailManagers = 150;
+        static constexpr int ExhaustFanDecoupledFromAvailManagers = 151;

```

Comment:
I didn't remove these, but I would suggest you doing it.  I made an enum class from the constant integers in the exterior energy use namespace.  The enum class should live back in the original header file.  You'll need to go in and clean up the usage of those variables but it is generally easy.  If you don't feel like tackling that during any of these refactors, it is OK, but don't bring them into this namespace.  They are really just constants, not variables.